tests/test_caching.py

```diff
@@ -684,14 +684,21 @@ def test_sparse_function(self, operate_on_empty_cache):
         # in the first clear_cache they were still referenced by their "parent" objects
         # (e.g., ru* by ConditionalDimensions, through `condition`)
 
-        assert len(_SymbolCache) == init_cache_size + 8
+        assert len(_SymbolCache) == init_cache_size + 12
         clear_cache()
         # Now we should be back to the original state except for
-        # pos* that belong to the abstract class
-        assert len(_SymbolCache) == init_cache_size + 2
+        # pos* that belong to the abstract class and the dimension
+        # bounds (x_m, x_M, y_m, y_M)
+        assert len(_SymbolCache) == init_cache_size + 6
         clear_cache()
-        # Now we should be back to the original state
-        assert len(_SymbolCache) == init_cache_size
+        # Now we should be back to the original state plus the dimension bounds
+        # (x_m, x_M, y_m, y_M)
+        assert len(_SymbolCache) == init_cache_size + 4
+        # Delete the grid and check that all symbols are subsequently garbage collected
+        del grid
+        for n in (10, 3, 0):
```

Comment:
Is 10 `init_cache_size` ? If so just 
```
del grid
assert len(_SymbolCache) == init_cache_size
```

And everything was deleted that's that's needed to check

Reply:
Init cache size is 11 (and consists of a different set of symbols)